_episodes/18.md

```diff
@@ -0,0 +1,560 @@
+the view query as the user that created the view, or as the `INVOKER`, which
+indicates to run the view query as the user that is running the outer query of
+the view. The default mode is `DEFINER`. See more 
+[here](https://trino.io/docs/current/sql/create-view.html#security).
+
+There are two types of views; materialized and logical views. The view defined
+above is the standard logical view that gets expanded into its definition. 
+Logical views don't provide any performance benefit since the data is not stored
+upfront and instead queried at query time. Materialized views persist the view 
+data upon view creation by storing the query data on disk.
+
+Schemas for logical views can't be quickly referenced as was done for the table 
+using `SHOW CREATE TABLE`. The query engine can quickly derive each of these
+types. The `SHOW CREATE VIEW` only shows the original query that was used to 
+create the view for logical tables but will show the actual schema of the
+materialized data.
+
+Materialized views make overall queries much faster to run as part of the query
+has already been computed and all that is required is recall. One issue with 
+materialized views is that the data may become outdated or new data may become
+available in the original tables. These tables can periodically be updated using
+a special refresh command or periodically have this procedure run automatically. 
+
+### Trino Views: So many views, so little time
+
+Views handling in Trino depends on the connector. In general, most connectors
+expose views to Trino as if they are another set of tables available for Trino
+to query. The main exceptions for this is the Hive and Iceberg connectors, which
+require Trino to do more as Trino is the primary query engine for these
+connectors. This simplifies thinking about how Trino handles views in general.
+The table below lists the current possible Hive and Iceberg views. 
+
+<table>
+<thead>
+  <tr>
+    <th colspan="2"></th>
+    <th>Logical</th>
+    <th>Materialized</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td rowspan="2">Trino Created View</td>
+    <td>Hive Connector</td>
+    <td>✅</td>
+    <td>❌</td>
+  </tr>
+  <tr>
+    <td>Iceberg Connector</td>
+    <td>❌</td>
+    <td>✅</td>
+  </tr>
+  <tr>
+    <td colspan="2">Hive Created View</td>
+    <td>✅ (read-only)</td>
+    <td>✅ (read-only)</td>
+  </tr>
+</tbody>
+</table>
+
+You'll notice that the Materialized views cannot be created through the Hive
+connector in Trino. You will get the following exception:
+
+```
+Caused by: java.sql.SQLException: Query failed (#...): 
+This connector does not support creating materialized views.
+```
+Also, you cannot create logical views in Iceberg and you will get the following
+exception:
+
+```
+Caused by: java.sql.SQLException: Query failed (#...): 
+This connector does not support creating views.
+```
+
+#### Trino Reads Hive Views
+
+Before Trino there was Hive. For many, Trino was the replacement for their Hive
```

Comment:
Trino is a replacement for the Hive runtime for many users, and it is very useful for these users to also be able to read data from Hive views in Trino.